### PR TITLE
Python wrappers for simplex and mixed meshes

### PR DESCRIPTION
**Pull request by agrayver**
Now you can query geometry and reference cell type from within python. Additionally, adds a wrapper for `GridGenerator::convert_hypercube_to_simplex_mesh`.

Related to #11508

### CODE REVIEW

**Review by peterrum — COMMENTED** [review]
Looks good to me! Thanks a lot!

**Comment by agrayver** [comment]
An example...

Firefox_Screenshot_2021-01-18T20-50-38 655Z

**Comment by peterrum** [comment]
FYI @eliasstudiert

File: contrib/python-bindings/include/reference_cell_wrapper.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2020 by the deal.II authors
```
Comment (Rombur) [inline contrib/python-bindings/include/reference_cell_wrapper.h:3]: Wrong years

File: contrib/python-bindings/source/cell_accessor_wrapper.cc [hunk 2]
```diff
@@ -756,6 +757,56 @@ namespace python
...
+  CellAccessorWrapper::n_faces() const
+  {
+    if ((dim == 2) && (spacedim == 2))
+      return internal::cell_cast<2, 2>(cell_accessor)->n_faces();
+    else if ((dim == 2) && (spacedim == 3))
+      return internal::cell_cast<2, 3>(cell_accessor)->n_faces();
+    else
+      return internal::cell_cast<3, 3>(cell_accessor)->n_faces();
+  }
+
```
Comment (Rombur) [inline contrib/python-bindings/source/cell_accessor_wrapper.cc:798]: Need more blank lines

File: contrib/python-bindings/source/export_reference_cell.cc [hunk 3]
```diff
@@ -0,0 +1,50 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2020 by the deal.II authors
```
Comment (Rombur) [inline contrib/python-bindings/source/export_reference_cell.cc:3]: Wrong years.

File: contrib/python-bindings/source/reference_cell_wrapper.cc [hunk 4]
```diff
@@ -0,0 +1,55 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2020 by the deal.II authors
```
Comment (Rombur) [inline contrib/python-bindings/source/reference_cell_wrapper.cc:3]: Years

File: contrib/python-bindings/source/reference_cell_wrapper.cc [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+DEAL_II_NAMESPACE_OPEN
+
+namespace python
+{
+  CellTypeWrapper::CellTypeWrapper(const CellTypeWrapper &other)
+  {
+    cell_type = other.cell_type;
+  }
+
+
```
Comment (Rombur) [inline contrib/python-bindings/source/reference_cell_wrapper.cc:29]: Not enough blank space between functions. Same below.

File: contrib/python-bindings/source/triangulation_wrapper.cc [hunk 6]
```diff
@@ -1534,6 +1549,25 @@ namespace python
   }
 
 
+  void
```
Comment (Rombur) [inline contrib/python-bindings/source/triangulation_wrapper.cc:1552]: Not enough blank lines

File: contrib/python-bindings/source/tria_accessor_wrapper.cc [hunk 7]
```diff
@@ -400,6 +408,45 @@ namespace python
...
 
+
+
+  unsigned int
+  TriaAccessorWrapper::n_vertices() const
+  {
+    if ((dim == 2) && (spacedim == 2) && (structdim == 1))
+      return internal::tria_accessor_cast<1, 2, 2>(tria_accessor)->n_vertices();
+    else if ((dim == 2) && (spacedim == 3) && (structdim == 1))
+      return internal::tria_accessor_cast<1, 2, 3>(tria_accessor)->n_vertices();
```
Comment (Rombur) [inline contrib/python-bindings/source/tria_accessor_wrapper.cc:419]: Why don't we have `internal::tria_accessor_cast<2, 2, 3>(tria_accessor)->n_vertices();`

File: contrib/python-bindings/source/tria_accessor_wrapper.cc [hunk 8]
```diff
@@ -400,6 +408,45 @@ namespace python
...
+    else
+      return internal::tria_accessor_cast<2, 3, 3>(tria_accessor)->n_vertices();
```
Comment (Rombur) [inline contrib/python-bindings/source/tria_accessor_wrapper.cc:421]: Why don't we have `internal::tria_accessor_cast<1, 3, 3>(tria_accessor)->n_vertices();` and `internal::tria_accessor_cast<3, 3, 3>(tria_accessor)->n_vertices();`?

**Review by Rombur — COMMENTED** [review]
The code looks fine but it would be nice to have a test.

**Comment by agrayver** [comment]
@Rombur addressed your comments, thanks.

File: contrib/python-bindings/source/export_reference_cell.cc [hunk 9]
```diff
@@ -0,0 +1,50 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2021 by the deal.II authors
```
Comment (Rombur) [inline contrib/python-bindings/source/export_reference_cell.cc:3]:
```suggestion
// Copyright (C) 2021 by the deal.II authors
```

File: contrib/python-bindings/source/reference_cell_wrapper.cc [hunk 10]
```diff
@@ -0,0 +1,64 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2021 by the deal.II authors
```
Comment (Rombur) [inline contrib/python-bindings/source/reference_cell_wrapper.cc:3]:
```suggestion
// Copyright (C) 2021 by the deal.II authors
```

**Review by Rombur — CHANGES_REQUESTED** [review]
I have a few more comments. I should have been paying more attention the first time around. Sorry.

**Comment by agrayver** [comment]
I fixed issues and added more cases. I suggest that the remaining cases are added on demand.

**Review by Rombur — APPROVED** [review]

**Pull request merged**